### PR TITLE
Release policy

### PR DESCRIPTION
**Pull request by Grummfy**
* adding page about release policy (new one)
* change configuration of jekyll to have a pages collection, this will make it clearer ;)
* add the permalink configuration for the posts, to be sure it's properly handled and will not change in the future
* add a contact page

see https://github.com/atoum/atoum.github.io/issues/16

### CODE REVIEW

File: _pages/about/release_policy.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+layout: static
+title: Release policy
+---
+
+# Release policy
+As said during the Forum PHP 2015, we should give our users a clear view of how/when we release atoum.
```
Comment (Hywan) [inline _pages/about/release_policy.md]: Link? Is it even necessary to say that?

File: _pages/about/release_policy.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Here are some notes I have on this:
```
Comment (Hywan) [inline _pages/about/release_policy.md]: “Here are the rules we agreed on and we follow”?

File: _pages/about/release_policy.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- **strictly stick to semver** (x++ if BC breaks, y++ for any new feature, z++ for any bug-fix)
```
Comment (Hywan) [inline _pages/about/release_policy.md]: Link to semver?

File: _pages/about/release_policy.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+- 1 release **every 6 weeks** (minor)
+- release **ASAP for any bug** (patch)
```
Comment (Hywan) [inline _pages/about/release_policy.md]: Unfold the acronym: ASAP -> as soon as possible.

File: _pages/about/release_policy.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+- the oldest will have its own branch and the newest will live on the master branch
+- bug-fixes will be backported
+- features will be backported if possible
+
+## Extensions (`atoum/*-extension`) / Stubs
+- will be released in-sync with atoum to conform to breaking changes (new atoum major release)
+- may be released at the same time (minor)
+- released ASAP for any bug-fix
+
+@atoum/rms feel free to comment on this, suggest new entry or fix some of them. Once we are OK with those rules, I'll write a blog post.
```
Comment (Hywan) [inline _pages/about/release_policy.md]: You must say that —by default— atoum is rolling-released. But to fit in real world constraints, we have adopted semver. So by default, `master` is always stable and represent the latest version. This is important to notice at first.

**Review by Hywan — CHANGES_REQUESTED** [review]

**Comment by Hywan** [comment]
Thanks for taking time for that!

File: _pages/about/release_policy.md (hunk 1 above)
Comment (Grummfy) [reply _pages/about/release_policy.md]: for now I just copy what is inside https://github.com/atoum/atoum.github.io/issues/16 I have not  work on the for yet

**Comment by Grummfy** [comment]
@Hywan, @mikaelrandy & @jubianchi I would like to have your though on the content of this new page.

**Review by mikaelrandy — COMMENTED** [review]
I agree on this, but i ask myself if we should add a RELEASE.md on atoum/atoum explaining this.

**Review by mikaelrandy — APPROVED** [review]

File: _pages/about/release_policy.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+### Deprecation
+
+Features may be marked as deprecated at any time in any minor version: this should at worst display a notice
+but the feature should be kept fully functional. The notice must contain hints on the alternatives. Once we start
+the new major branch deprecated features will be removed.
+
+## Support of previous major version
+
+We try to support a previous major version until *one* year after the *last minor/patch* release.
```
Comment (jubianchi) [inline _pages/about/release_policy.md]: `s/We try/We will/`

File: _pages/about/release_policy.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+### Multiple major releases living side-by-side
+
+When we release a new major version, we only support the current and the previous major version, but not more. To handle that,
+we follow this process:
+* the oldest major version will have its own branch and the newest will live on the master branch
```
Comment (jubianchi) [inline _pages/about/release_policy.md]: add a `;` at the end

File: _pages/about/release_policy.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+* bug-fixes will be backported
```
Comment (jubianchi) [inline _pages/about/release_policy.md]: add a `;` at the end

File: _pages/about/release_policy.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+* features will be backported if possible or asked.
+
+But, for any reason next major may start before the previous one reaches EOL.
+
+In any case next minor/patch release source code lives on the master branch.
+
+## Semver & rolling-release
+atoum follow semver, but also [rolling release](https://en.wikipedia.org/wiki/Rolling_release) principle. So
+the *master* branch is considered as stable and can be directly used without any issue and represent the latest stable version.
```
Comment (jubianchi) [inline _pages/about/release_policy.md]: I would have used bold for `the *master* branch is considered as stable`

File: _pages/about/release_policy.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+But, for any reason next major may start before the previous one reaches EOL.
+
+In any case next minor/patch release source code lives on the master branch.
+
+## Semver & rolling-release
+atoum follow semver, but also [rolling release](https://en.wikipedia.org/wiki/Rolling_release) principle. So
+the *master* branch is considered as stable and can be directly used without any issue and represent the latest stable version. 
+
+## Extensions
+The officially supported extension[^1] has a different release policy. It's more released when it's ready:
```
Comment (jubianchi) [inline _pages/about/release_policy.md]:
```diff
-supported extension
+supported extensions
```

```diff
- It's more released when it's ready
+ They are released when they are ready
```

File: _pages/about/release_policy.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+* if there is BC break, will be released in-sync with atoum for new atoum major release
```
Comment (jubianchi) [inline _pages/about/release_policy.md]:
```diff
- if there is BC break, will be
+ if there is any BC break in atoum, they will be
```

Comment (jubianchi) [reply _pages/about/release_policy.md]: add a `;` at the end

File: _pages/about/release_policy.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+* may be released at the same time (minor)
```
Comment (jubianchi) [inline _pages/about/release_policy.md]: add a ; at the end

**Review by jubianchi — CHANGES_REQUESTED** [review]

File: _pages/about/release_policy.md (hunk 6 above)
Comment (mikaelrandy) [reply _pages/about/release_policy.md]: `we will try` or `we will support` ?

Comment (jubianchi) [reply _pages/about/release_policy.md]: indeed:

```diff
- We try to support
+ We will support
```

**Comment by Grummfy** [comment]
I have made the fix from @jubianchi . I also edit the source page to remove release process from it and merge it into the release policy page. I also add a contact page.

For me, now it's finished.

**Pull request merged**